From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kevin Hilman <khilman@baylibre.com>
Date: Thu, 4 Feb 2021 12:45:39 -0800
Subject: ANDROID: GKI: Kconfig.gki: add hidden PHY configs

When testing GKI with out-of-tree PCIe support as modules, it was
discovered some hidden Kconfigs are required for DPHYs.  Add here so
out of tree PCIe drivers and PHYs can be built in a separate tree.

Bug: 179406580
Change-Id: Ic27c678b4e0dd4c29d354ff1139d453b99a3744f
Signed-off-by: Kevin Hilman <khilman@baylibre.com>
---
 init/Kconfig.gki | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -193,6 +193,13 @@ config GKI_HIDDEN_NET_CONFIGS
 	  Dummy config option used to enable the networking hidden
 	  config, required by various SoC platforms.
 
+config GKI_HIDDEN_PHY_CONFIGS
+	bool "Hidden PHY configuration needed for GKI"
+	select GENERIC_PHY_MIPI_DPHY
+	help
+	  Dummy config option used to enable the hidden PHY configs,
+	  required by various SoC platforms.
+
 
 # Atrocities needed for
 # a) building GKI modules in separate tree, or
@@ -221,6 +228,7 @@ config GKI_HACKS_TO_FIX
 	select GKI_HIDDEN_IRQ_CONFIGS
 	select GKI_HIDDEN_HYPERVISOR_CONFIGS
 	select GKI_HIDDEN_NET_CONFIGS
+	select GKI_HIDDEN_PHY_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
